src/Summoner/Config.hs

```diff
@@ -0,0 +1,170 @@
+    (<>) = gsappenddefault
+
+instance Monoid PartialConfig where
+    mempty = gmemptydefault
+    mappend = (<>)
+
+-- | Default 'Config' configurations.
+defaultConfig :: PartialConfig
+defaultConfig = Config
+    { cOwner    = Last (Just "kowainik")
+    , cFullName = Last (Just "Kowainik")
+    , cEmail    = Last (Just "xrom.xkov@gmail.com")
+    , cLicense  = Last (Just $ License "MPL-2.0")
+    , cGitHub   = Idk
+    , cTravis   = Idk
+    , cAppVey   = Idk
+    , cPrivate  = Idk
+    , cScript   = Idk
+    , cLib      = Idk
+    , cExe      = Idk
+    , cTest     = Idk
+    , cBench    = Idk
+    , cGhcVer   = Last (Just [])
+    }
+
+-- | Identifies how to read 'Config' data from the @.toml@ file.
+configT :: BiToml PartialConfig
+configT = Config
+    <$> lastP Toml.str "owner"       .= cOwner
+    <*> lastP Toml.str "fullName"    .= cFullName
+    <*> lastP Toml.str "email"       .= cEmail
+    <*> lastP license  "license"     .= cLicense
+    <*> decision       "github"      .= cGitHub
+    <*> decision       "travis"      .= cTravis
+    <*> decision       "appveyor"    .= cAppVey
+    <*> decision       "private"     .= cPrivate
+    <*> decision       "bscript"     .= cScript
+    <*> decision       "lib"         .= cLib
+    <*> decision       "exe"         .= cExe
+    <*> decision       "test"        .= cTest
+    <*> decision       "bench"       .= cBench
+    <*> lastP (Toml.arrayOf ghcVerV) "ghcVersions" .= cGhcVer
+  where
+    lastP :: (Key -> BiToml a) -> Key -> BiToml (Last a)
+    lastP f = dimapBijection getLast Last . Toml.maybeP f
+
+    ghcVerV :: Valuer 'TString GhcVer
+    ghcVerV = Valuer (matchText >=> parseGhcVer) (Toml.String . showGhcVer)
+
+    license :: Key -> BiToml License
+    license =  bijectionMaker "License" (matchText >=> Just . License) (Toml.String . unLicense)
+
+    decision :: Key -> BiToml Decision
+    decision = dimapBijection fromDecision toDecision . Toml.maybeP Toml.bool
+
+    decisionMaybe :: [(Decision, Maybe Bool)]
+    decisionMaybe = [ (Idk, Nothing)
+                    , (Yes, Just True)
+                    , (Nop, Just False)
+                    ]
+
+    fromDecision :: Decision -> Maybe Bool
+    fromDecision d = snd $ head $ filter ((== d) . fst) decisionMaybe
+
+    toDecision :: Maybe Bool -> Decision
+    toDecision m = fst $ head $ filter ((== m) . snd) decisionMaybe
+
+
+-- | Make sure that all the required configurations options were specified.
+finalise :: PartialConfig -> Either Text Config
+finalise Config{..} = Config
+    <$> fin  "owner"    cOwner
+    <*> fin  "fullName" cFullName
+    <*> fin  "email"    cEmail
+    <*> fin  "license"  cLicense
+    <*> pure cGitHub
+    <*> pure cTravis
+    <*> pure cAppVey
+    <*> pure cPrivate
+    <*> pure cScript
+    <*> pure cLib
+    <*> pure cExe
+    <*> pure cTest
+    <*> pure cBench
+    <*> fin  "ghcVersions" cGhcVer
+  where
+    fin name = maybe (Left $ "Missing field: " <> name) Right . getLast
```

Comment:
Actually, this is an interesting piece of code. `Applicative` instance for `Either` works in that way that it fails on first error and returns only single error to user. But it would be much more convenient to tell all absent fields at once. This actually already exist in Haskell world. There's data type called [`Validation`](https://hackage.haskell.org/package/either-5/docs/Data-Either-Validation.html#t:Validation). It's like `Either` (just different constructor names). But it has slightly more interesting [`applicative instance`](https://hackage.haskell.org/package/either-5/docs/src/Data.Either.Validation.html#line-57). It accumulates all errors. So we can copy-paste `Validation` to us and this instance and just use it here instead of `Either` to output multiple errors to user regarding absent fields!